src/Drivers/Zibal/Zibal.php

```diff
@@ -108,6 +108,51 @@ public function purchase()
             $data = array_merge($data, $allowedCards);
         }
 
+        // Checking if optional percentMode parameter exists
+        $percentMode = null;
+        if (!empty($details['percentMode'])) {
+            $percentMode = $details['percentMode'];
+        } elseif (!empty($this->settings->percentMode)) {
+            $percentMode = $this->settings->percentMode;
+        }
+
+        if ($percentMode != null) {
+            $percentMode = array(
+                'percentMode' => $percentMode,
+            );
+            $data = array_merge($data, $percentMode);
+        }
+
+        // Checking if optional feeMode parameter exists
+        $feeMode = null;
+        if (!empty($details['feeMode'])) {
+            $feeMode = $details['feeMode'];
+        } elseif (!empty($this->settings->feeMode)) {
+            $feeMode = $this->settings->feeMode;
+        }
+
+        if ($feeMode != null) {
+            $feeMode = array(
+                'feeMode' => $feeMode,
+            );
+            $data = array_merge($data, $feeMode);
+        }
+
+        // Checking if optional multiplexingInfos parameter exists
+        $multiplexingInfos = null;
```

Comment:
and these checks can be done in another method. why do you like to write all the logic in a single method and make it huge? separate your concerns and break them into methods to be more meaningful and easy to read.

Reply:
You are right. I was trying not to change so much in code but it would be cleaner to add the methods you mentioned. I created two new methods. One to extract the detail using detail name sent and one to check for optional details and add them to the sent data if they are not null. This way adding new parameters should be easier and cleaner in the future.